infra/gcp/lib_iam.sh

```diff
@@ -184,32 +235,139 @@ function ensure_removed_project_role_binding() {
     _ensure_removed_resource_role_binding "projects" "${project}" "${principal}" "${role}"
 }
 
-# Ensure that IAM binding has been removed from organization
+# Ensure that custom IAM role exists in scope and in sync with definition in file
 # Arguments:
-#   $1:  The principal (e.g. "group:k8s-infra-foo@kubernetes.io")
-#   $2:  The role name (e.g. "roles/foo.bar")
-function ensure_removed_org_role_binding() {
-    if [ ! $# -eq 2 -o -z "$1" -o -z "$2" ]; then
-        echo "ensure_removed_org_role_binding(principal, role) requires 2 arguments" >&2
+#   $1:  The scope of the role (e.g. "organization", "project")
+#   $2:  The id of the scope (e.g. "12345819", "k8s-infra-foo")
+#   $3:  The role name (e.g. "foo.barrer")
+#   $4:  The file (e.g. "/path/to/file.yaml")
+function _ensure_custom_iam_role_from_file() {
+    if [ ! $# -eq 4 -o -z "$1" -o -z "$2" -o -z "$3" -o -z "$4" ]; then
+        echo "_ensure_custom_iam_role_from_file(scope, id, name, file) requires 4 arguments" >&2
         return 1
     fi
 
-    local organization="${GCP_ORG}"
-    local principal="${1}"
-    local role="${2}"
+    local scope="${1}"
+    local id="${2}"
+    local name="${3}"
+    local file="${4}"
+    case "${scope}" in
+      organization | project ) ;;
+      * )
+        echo >/dev/stderr "_ensure_custom_iam_role_from_file(scope, id, name, file) scope must 'organization' or 'project'"
```

Comment:
I forget the details, but there was something that caused me to switch everything to 

```
echo "_ensure_custom_iam_role_from_file(scope, id, name, file) scope must 'organization' or 'project'" >&2
```

syntax

Reply:
done